crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -43,6 +43,36 @@ impl SystemMeta {
         &self.name
     }
 
+    /// Returns the system's component access set.
+    #[inline]
+    pub fn component_access_set(&self) -> &FilteredAccessSet<ComponentId> {
+        &self.component_access_set
+    }
+
+    /// Returns a mutable reference to the system's component access set.
+    ///
+    /// # Safety
+    /// This allows unsafe modifications to the component access set that can violate Bevy's scheduling soundness.
+    #[inline]
+    pub unsafe fn component_access_set_mut(&mut self) -> &mut FilteredAccessSet<ComponentId> {
```

Comment:
The `pub(crate)` on the fields should be removed and the `bevy_ecs` uses of the access sets should use these accessors.